src/big/big_int.cr

```diff
@@ -390,6 +390,14 @@ struct BigInt < Int
     LibGMP.popcount(self)
   end
 
+  def leading_zeros_count
```

Comment:
I'd prefer to simply not declare this method on `BigInt`. It does only apply to fixed-width integers.

Reply:
Agreed, this is just repeating compiler's work.

Reply:
That's impossible since it's declared as `abstract`.

Reply:
Yes, it shouldn't be an abstract method on `Int`, because the concept of leading zeros doesn't apply to all integer types.

Reply:
@straight-shoota so what do you suggest?

Reply:
Drop the `Int#leading_zeros_count` abstract method.

Reply:
> Yes, it shouldn't be an abstract method on `Int`, because the concept of leading zeros doesn't apply to all integer types.

For the time being it applies to all of the `Int` types, besides `BigInt` so I'm more inclined to treat `BigInt` as an exception to the rule then the other way around.

Besides, it's symmetric to `Int#trailing_zeros_count`. What's exactly the problem with raising in compile-time in `BigInt#leading_zeros_count` impl.?

Reply:
It declares a method that makes no sense, and duplicates what the compiler already does.

Reply:
Yeah, but having it declared directly on `Int` as `abstract` makes it possible to use it for all `Int` types without duck typing or any other checks.

Reply:
It's the other way around: if we ever support holding a variable of type Int, then invoking a method in it will have the compiler type all methods of all subclasses. It will find the one in BigInt and the compilation will fail.

Reply:
Oh, I've forgot about that case, I'll remove it then. Is it ok to just comment it out, to keep it more visible?

Reply:
Done.